guardian/recipe-tools
# Add migration scripts

## What does this change?

Adds a script to apply a filter, modification and output JQ expression to a recipe, to enable batch partial updates to recipes.

Example use:

```
./transform-recipes.sh \\
  -n commas-and-whitespace \\
  -f 'select(.ingredients[].ingredientsList[].name | test(\"(,| )+$\")) | .' \\
  -q '.ingredients[].ingredientsList[].name |= sub(\"(?<ingredient>.*?)[, ]+$\"; \"\(.ingredient)\")' \\
  -o '{ id, ingredients }' \\
  -d false
```

See the script help `-h` for more detail.

NB: I wanted to write this in Rust, but I didn't have the time to get the ergonomics I felt necessary for rapid iteration: for example, a comprehensive diff of all the changes while dry-running, or quick iteration on modifications, which `jq` really helps with. Be up for a rewrite when we're under less time pressure!

At the moment this script relies on a local corpus of recipes generated by the Rust script and available in the `recipes-scripts` folder at `recipes.ndjson`.

## How to test

Given the core recipe corpus, run the script, either in dry-run mode or against a local or CODE endpoint using [this branch of flexible-content](https://github.com/guardian/flexible-content/pull/4643). Does it perform as expected?

Tested locally and against CODE. The migrations themselves look sensible but are not yet comprehensively tested – we'll iterate as we test against some CODE content.
